lib/scanner/walk.go

```diff
@@ -131,13 +136,59 @@ func (w *walker) walk() (chan protocol.FileInfo, error) {
 	// been modified to the counter routine.
 	go func() {
 		hashFiles := w.walkAndHashFiles(toHashChan, finishedChan)
+
+		var dirs []string
 		if len(w.Subs) == 0 {
-			filepath.Walk(w.Dir, hashFiles)
+			// The list of dirs to walk is the folder dir itself.
+			dirs = []string{w.Dir}
 		} else {
+			// If a set of subdirs was given, replace the list with the list of
+			// subdirs, relative to the folder path.
 			for _, sub := range w.Subs {
-				filepath.Walk(filepath.Join(w.Dir, sub), hashFiles)
+				path := filepath.Join(w.Dir, sub)
+				dirs = append(dirs, path)
+			}
+		}
+
+		// If a there's a set of symlinks to follow, do the same for those.
+		if len(w.FollowSymlinks) > 0 {
+		nextSymlink:
+			for _, link := range w.FollowSymlinks {
+				path := filepath.Join(w.Dir, link)
+
+				// Verify that the symlink is under one of the dirs we
+				// intend to scan.
+				for _, allowed := range dirs {
+					if strings.HasPrefix(path, allowed+string(os.PathSeparator)) {
+						goto ok
+					}
+				}
+				continue nextSymlink
```

Comment:
Don't understand this whole magic.
I think the FollowSymlinks should be existing symlinks that are relative to the folder root.